error: invalid argument '-std=gnu++2b' not allowed with 'C'

libhyprcursor/hyprcursor.cpp:471:36: error: arithmetic on a pointer to void
  471 |     std::memcpy(output, DATA->data + DATA->readNeedle, toRead);
      |                         ~~~~~~~~~~ ^

--- CMakeLists.txt.orig	2024-03-08 17:32:40 UTC
+++ CMakeLists.txt
@@ -44,12 +44,6 @@ target_link_libraries(hyprcursor PkgConfig::deps)
 
 target_link_libraries(hyprcursor PkgConfig::deps)
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    # for std::expected.
-    # probably evil. Arch's clang is very outdated tho...
-    target_compile_options(hyprcursor PUBLIC -std=gnu++2b -D__cpp_concepts=202002L -Wno-macro-redefined)
-endif()
-
 # hyprcursor-util
 add_subdirectory(hyprcursor-util)
 
--- libhyprcursor/hyprcursor.cpp.orig	2024-03-08 17:32:40 UTC
+++ libhyprcursor/hyprcursor.cpp
@@ -468,7 +468,7 @@ static cairo_status_t readPNG(void* data, unsigned cha
 
     size_t toRead = len > DATA->dataLen - DATA->readNeedle ? DATA->dataLen - DATA->readNeedle : len;
 
-    std::memcpy(output, DATA->data + DATA->readNeedle, toRead);
+    std::memcpy(output, (uint8_t*)DATA->data + DATA->readNeedle, toRead);
     DATA->readNeedle += toRead;
 
     if (DATA->readNeedle >= DATA->dataLen) {
